### ISSUE TITLE
Conflicting class docstring rules D203 and D211

### ISSUE THREAD

**Issue by mbdevpl** [PyCQA/pydocstyle#400]
First of all, unserstanding this issue requires familiarity with PEP8, PEP257, as well as this commit (from year 2014): https://hg.python.org/peps/rev/9b715d8246db which changed PEP257 in a mysterious way.

Let me share a curious occurence in a following code (for simplicity I skip surrounding code):
```python3
...

class Folder(t.List[Message]):

    """For storing messages.

    Extends the list in order to store messages.
    """

    def __init__(self, name: str, *args, **kwargs):
        ...
```

And then, I run `pycodestyle` (skipping other results for simplicity):
```
$ pydocstyle maildaemon/folder.py
maildaemon/folder.py:13 in public class `Folder`:
        D211: No blank lines allowed before class docstring (found 1)
```

And then, I delete the blank line that causes D211, only to get this (skipping other results for simplicity):
```
$ pydocstyle maildaemon/folder.py
maildaemon/folder.py:12 in public class `Folder`:
        D203: 1 blank line required before class docstring (found 0)
```

Therefore rules D203 and D211 conflict with each other.

In any case, the official Python docs (at this point in time) don't clearly recommend one way or the other, but the comment in the commit mentioned above states "Do not recommend a blank line before any docstrings." so the trend seems to be towards recommending no new-lines before class docstrings, which at least is consistent with "no new-lines before function docstrings" rule.

To sum up, I think one (or both) of the rules mentioned above need(s) to be removed from pydocstyle. From how I see the what's going on in the Python community, I'd recommend removing D203 and keeping D211, but have no strong opinion one way or the other.

**Comment by mbdevpl** [comment]
Ok, nevermind, actually I kept reading and reading and reading and eventually I found that this is by design and I should tweak my settings. Sorry for the noise.
